src/components/common/Select.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+
+export default function Select({ className, onSelect, options = [], value, ...rest }) {
+    const [selected, setSelected] = useState(options.find(option => option.value === value) || options[0]);
+    const [selectOpen, setSelectOpen] = useState(false);
+
+    useEffect(() => {
+        setSelected(options.find(option => option.value === value));
+    }, [value]);
+
+    const toggleDropdown = () => setSelectOpen(!selectOpen);
+
+    const selectFromDropdown = option => {
+        setSelected(option);
+        toggleDropdown();
+        if (onSelect) {
+            onSelect(option);
+        }
+    };
+
+    const selectClassnames = `relative rounded ${className}`;
+
+    return (
+        <div className={selectClassnames} {...rest}>
+            <div
+                className={`flex cursor-pointer items-center justify-between button-outline px-4 py-2 ${
+                    selectOpen && 'rounded-b-none'
+                }`}
+                onClick={toggleDropdown}
+            >
+                {selected.icon && <img className='mr-2' src={selected.icon} alt={selected.label} />}
+                <p className='grow mr-2'>{selected.label}</p>
+                <span className='icon-base'>{selectOpen ? 'keyboard_arrow_up' : 'keyboard_arrow_down'}</span>
+            </div>
+            {selectOpen && (
+                <div className='absolute bg-primary content-outline rounded-b w-full z-50'>
+                    {options.map(option => (
+                        <div
+                            className='flex items-center cursor-pointer px-4 py-2 last:rounded-b odd:bg-secondary bg-hover-subdued'
```

Comment:
The odd rows aren't getting the hover color here for some reason. (also - maybe subjective but the background on the odd rows seems a little dark, thoughts on making them match the colors on the calc table rows which are a bit lighter?)

Reply:
Updated to match table row colours and fixed the hover color, a downside of tailwinds `odd:` and css-specificity though 😞